blivet/devicetree.py

```diff
@@ -203,6 +203,12 @@ def _remove_device(self, dev, force=None, modparent=True):
                        device.disk == dev.disk:
                         device.update_name()
 
+        if isinstance(dev, LVMLogicalVolumeDevice) and dev.from_lvs:
+            # if removing an LV constructed from other LVs, we need to put the
+            # LVs it's supposed to be constructed from back into the device tree
+            for lv in dev.from_lvs:
+                self._add_device(lv, new=False)
```

Comment:
I'm not sure about this. It makes more sense to me to make sure that whenever the device is added/removed it takes care about its components no matter if it is an action cancellation or (possibly) something else.